src/ape/types/trace.py

```diff
@@ -167,35 +166,25 @@ def get_gas_report(self, exclude: Optional[List["ContractFunctionPath"]] = None)
         """
 
         exclusions = exclude or []
-
-        for exclusion in exclusions:
-            if exclusion.method_name is None and fnmatch(self.contract_id, exclusion.contract_name):
-                # Skip this whole contract. Search contracts from sub-calls.
-                return merge_reports(*(c.get_gas_report(exclude) for c in self.calls))
-
-            for excl in exclusions:
-                if not excl.method_name:
-                    # Full contract skips handled above.
-                    continue
-
-                elif not fnmatch(self.contract_id, excl.contract_name):
-                    # Method may match, but contract does not match, so continue.
-                    continue
-
-                elif self.method_id and fnmatch(self.method_id, excl.method_name):
-                    # Skip this report because of the method name exclusion criteria.
-                    return merge_reports(*(c.get_gas_report(exclude) for c in self.calls))
-
-        reports = [c.get_gas_report(exclude) for c in self.calls]
-        if self.method_id:
-            report = {
-                self.contract_id: {
-                    self.method_id: [self.gas_cost] if self.gas_cost is not None else []
-                }
-            }
-            reports.append(report)
-
-        return merge_reports(*reports)
+        if (
+            not self.contract_id
+            or not self.method_id
+            or _exclude_gas(exclusions, self.contract_id, self.method_id)
+        ):
+            return merge_reports(*(c.get_gas_report(exclude) for c in self.calls))
+
+        elif self.method_id not in ["0x", *[str(i) for i in range(10)]]:
```

Comment:
This is for excluding precompiles right? I think a comment here mentioning that will save someone some time in the future